### PR TITLE
minor stylistic clippy cleanups

### PR DESCRIPTION
**Pull request by matthiaskrgr**
simplify if let Some(_) = x  to  if x.is_some()  (clippy::redundant_pattern_matching)
don't create owned values for comparison (clippy::cmp_owned)
use .contains() or .any() instead of find(x).is_some() (clippy::search_is_some)
don't wrap code block in Ok()  (clipppy::unit_arg)

### CODE REVIEW

File: compiler/rustc_incremental/src/persist/load.rs [hunk 1]
```diff
@@ -209,10 +209,7 @@ pub fn load_query_result_cache<'a>(
     sess: &'a Session,
     definitions: &Definitions,
 ) -> Option<OnDiskCache<'a>> {
-    if sess.opts.incremental.is_none() {
-        return None;
-    }
-
+    sess.opts.incremental.as_ref()?;
```
Comment (LingMan) [inline compiler/rustc_incremental/src/persist/load.rs:212]: I'd leave this part unchanged.
To me `?` conveys a meaning of  "I want to use this value but the container is in the way". Here the intent is about looking at the container itself, so the more explicit code makes it easier to understand IMO.

**Review by LingMan — COMMENTED** [review]
One hunk I'd drop (see inline comment) but the rest look like improvements to me.

File: compiler/rustc_incremental/src/persist/load.rs (hunk 1 above)
Comment (matthiaskrgr) [reply compiler/rustc_incremental/src/persist/load.rs]: dropped

**Pull request merged**